wolfcrypt/src/port/Renesas/renesas_sce_ra6m3g.c

```diff
@@ -0,0 +1,1146 @@
+        if (ret == MP_OKAY)
+            ret = mp_to_unsigned_bin(key->pubkey.y, &pub[keySz]);
+    }
+
+    /* Perform Hardware Key Generation */
+    if (ret == MP_OKAY && wolfSSL_CryptHwMutexLock() == 0)
+    {
+        switch (key->dp->id) {
+        case ECC_SECP256R1:
+        case ECC_SECP256K1:
+            ret = (int) HW_SCE_ECC_256VerifySign((const uint32_t*) domain,
+                                                 (const uint32_t*) gxy,
+                                                 (const uint32_t*) pub,
+                                                 (const uint32_t*) hash,
+                                                 (const uint32_t*) &sigRS[0],
+                                                 (const uint32_t*) &sigRS[keySz]);
+            break;
+        case ECC_SECP384R1:
+            ret = (int) HW_SCE_ECC_384VerifySign((const uint32_t*) domain,
+                                                 (const uint32_t*) gxy,
+                                                 (const uint32_t*) pub,
+                                                 (const uint32_t*) hash,
+                                                 (const uint32_t*) &sigRS[0],
+                                                 (const uint32_t*) &sigRS[keySz]);
+            break;
+        default:
+            WOLFSSL_MSG("Unsupported ECC Curve.");
+            ret = ECC_CURVE_OID_E;
+            break;
+        }
+        wolfSSL_CryptHwMutexUnLock();
+
+        if (ret == FSP_SUCCESS) {
+            *res = 1;
+            ret = MP_OKAY;
+        } else {
+            /* invalid signature, result should already be 0 */
+            if (ret == FSP_ERR_CRYPTO_SCE_VERIFY_FAIL) {
+                WOLFSSL_MSG("Verify failed.");
+                ret = MP_OKAY;
+            }
+        }
+
+    }
+    return ret;
+}
+
+/*
+   ECC Scalar Multiplication using Renesas RA Hardware
+   Supported Curves: ECC_SECP256R1, ECC_SECP256K1, ECC_SECP384R1
+
+   Acts as a HW replacement to wc_ecc_mulmod_ex
+   Note: The hardware performance is worse than using wolfSSL software
+         mulmod with WOLFSSL_HAVE_SP_ECC and WOLFSSL_SP_ARM_CORTEX_M_ASM.
+
+   Inputs:
+       k        The multiplicand
+       G        Base point to multiply
+       a        ECC curve parameter a
+       b        ECC curve parameter b
+       modulus  prime / modulus for the curve
+       map      [boolean] If non-zero maps the point back to affine coordinates,
+                 otherwise it's left in jacobian-montgomery form
+
+   Outputs:
+       R        Destination of product
+
+   return  MP_OKAY:         HW completed successfully
+           BAD_FUNC_ARG:    Invalid Argument
+           ECC_CURVE_OID_E: Unsupported curve
+*/
+int wc_Renesas_Ecc256Mulmod(mp_int* k, ecc_point *G, ecc_point *R,
+                            mp_int* a, mp_int* b, mp_int* modulus, int map) {
+
+    word32 keySize = 0;
+    int ret = MP_OKAY;
+    uint8_t k_bin[ECC384_KEYSIZE]    = {0};
```

Comment:
I changed that.  I used uint8_t since the Renesas HW drivers accept uint32_t and I was keeping the same convention.  I used uint8_t instead of uint32_t because I ensure that everything is 32-bit aligned and I thought byte sized buffers helped readability / syntax.